File: mvvm-java/domain/src/main/java/com/wasisto/githubuserfinder/domain/interfaces/GithubDataSource.java
package com.wasisto.githubuserfinder.domain.interfaces;

import com.wasisto.githubuserfinder.domain.models.SearchUsersResult;
import com.wasisto.githubuserfinder.domain.models.User;

public interface GithubDataSource {

    SearchUsersResult searchUsers(String query) throws Exception;

    User getUser(String username) throws Exception;
}

File: mvp-kotlin/domain/src/main/java/com/wasisto/githubuserfinder/domain/interfaces/GithubDataSource.kt
package com.wasisto.githubuserfinder.domain.interfaces

import com.wasisto.githubuserfinder.domain.models.SearchUsersResult
import com.wasisto.githubuserfinder.domain.models.User

interface GithubDataSource {

    fun searchUsers(query: String): SearchUsersResult

    fun getUser(username: String): User
}